interactive_templates/templates/v2/analysis/measures.py

```diff
@@ -14,6 +14,117 @@ def round_column(df, col, decimals=-1):
+        date (str): The date of the input file.
+        group (str, optional): The group category. Defaults to None.
+        group_value (str, optional): The group value. Defaults to None.
+    Returns:
+        pd.DataFrame: A DataFrame (shape: (1, 5)) containing the total counts.
+    """
+    if not {"event_measure", "population", "date"}.issubset(df.columns):
+        raise ValueError("The input DataFrame must contain 'event measure', 'population' and 'date' columns.")
+
+    count = df["event_measure"].sum()
+    population = df["population"].sum()
+
+    row_dict = {
+        "date": date,
+        "event_measure": count,
+        "population": population,
+        "group": group,
+        "group_value": group_value,
+    }
+    return pd.DataFrame.from_records([row_dict])
+
+
+def calculate_group_counts(df, breakdown, date):
+    """
+    Calculate the counts for a specified group.
+
+    Args:
+        df (pd.DataFrame): The input DataFrame. Should contain a column named "breakdown".
+        breakdown (str): The name of the column to group by.
+        date (str): The date of the input file.
+
+    Returns:
+        pd.DataFrame: A DataFrame containing the counts for the specified group.
+    """
+    counts = (
+        df.groupby(by=[breakdown])["event_measure"]
+        .agg(["sum", "count"])
+        .reset_index()
+        .rename(
+            columns={
+                breakdown: "group_value",
+                "sum": "event_measure",
+                "count": "population",
+            }
+        )
+    )
+    counts["date"] = date
+    counts["group"] = breakdown
+
+    return counts
+
+
+def calculate_and_redact_values(df):
+    """
+    Calculate the values for each group and redact where necessary.
+
+    Args:
+        df (pd.DataFrame): The input DataFrame. Should contain columns "event_measure", "population" and "group".
+
+    Returns:
+        pd.DataFrame: A DataFrame containing the calculated values.
+    """
+    groups = df["group"].unique()
+    result = pd.DataFrame(columns=["group", "group_value", "value"])
+    for group in groups:
+        group_df = df.loc[df["group"] == group, :]
+
+        if group == "practice":
+            group_df.loc[:, "value"] = calculate_rate(
+                group_df, "event_measure", "population"
+            )
+        else:
+            group_df = round_column(group_df, "event_measure", decimals=-1)
+            group_df = round_column(group_df, "population", decimals=-1)
+            group_df.loc[:, "value"] = calculate_rate(
+                group_df, "event_measure", "population"
+            )
+            group_df.loc[
+                (group_df["event_measure"] == 0) | (group_df["population"] == 0),
+                "value",
+            ] = "[Redacted]"
+
+        result = pd.concat([result, group_df], ignore_index=True)
+
+    return result
+
+
 def parse_args():
     parser = argparse.ArgumentParser()
```

Comment:
args.measure is currently unused? Could be removed from and the call in the project.yaml